src/matlab2tikz.m

```diff
@@ -204,6 +204,10 @@ function matlab2tikz(varargin)
     ipp = ipp.addParamValue(ipp, 'showWarnings', true, @islogical);
     ipp = ipp.addParamValue(ipp, 'checkForUpdates', isempty(VCID), @islogical);
 
+    ipp = ipp.addParamValue(ipp, 'semanticalLineWidth', false, @islogical);
+    ipp = ipp.addParamValue(ipp, 'semanticalValueList', [.1,.2,.4,.6,.8,1.2,1.6], @isnumeric);
```

Comment:
I like that you have these as input argument, however, I think the input format could be improved a bit.

Right now, validity of `semanticalValueList` and `semanticalStringList` is coupled. That is not ideal. What I propose instead is to use a `semanticalLineWidths` parameter instead similar to what we have for `extraColors`.

E.g.
``` matlab
ipp = ipp.addParamValue(ipp, 'semanticalLineWidths', {'ultra thin',0.1; 'very thin', 0.2; ...}, @isValidLineWidthMapping);
```

for which the `isValidLineWidthMapping` function could be something like:
```matlab
function bool = isValidLineWidthMapping(value)
bool = iscell(value) && size(value, 2) == 2; % Nx2 cell array
bool = bool && all(cellfun(@ischar   , value(:,1)); % first column: names
bool = bool && all(cellfun(@isnumeric, value(:,2)); % second column: line width in points
```

Because that is something that can be checked easily and you can always extract both lists you had from that data structure.